core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala

```diff
@@ -89,9 +100,148 @@ class ReplicaFetcherThreadTest {
       t1p1 -> new EpochEndOffset(Errors.NONE, UNDEFINED_EPOCH, UNDEFINED_EPOCH_OFFSET)
     )
 
+    assertFalse("Leader request should not have been sent", toFail)
+    assertEquals("results from leader epoch request should have undefined offset", expected, result)
+  }
+
+  /**
+    * If a partition doesn't have an epoch in the cache, this means it either
+    *   does not support epochs (log message format below 0.11) or it is a bootstrapping follower.
+    *
+    * In both cases, he has an undefined epoch
```

Comment:
nit: "he" -> "the partition"?

Personally, I've always thought of partitions as feminine. 😉

Reply:
Haha, love it. Thanks for catching that, I try to avoid gender-specific pronouns in code comments - I think it's a language-barrier issue